underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -84,16 +84,16 @@ public S3UnderFileSystem(String bucketName, TachyonConf tachyonConf) throws Serv
     mBucketName = bucketName;
 
     Jets3tProperties props = new Jets3tProperties();
-    if (tachyonConf.containsKey(Constants.S3_PROXY_HOST)) {
+    if (tachyonConf.containsKey(Constants.UNDERFS_S3_PROXY_HOST)) {
       props.setProperty("httpclient.proxy-autodetect", "false");
-      props.setProperty("httpclient.proxy-host", tachyonConf.get(Constants.S3_PROXY_HOST));
-      props.setProperty("httpclient.proxy-port", tachyonConf.get(Constants.S3_PROXY_PORT));
+      props.setProperty("httpclient.proxy-host", tachyonConf.get(Constants.UNDERFS_S3_PROXY_HOST));
+      props.setProperty("httpclient.proxy-port", tachyonConf.get(Constants.UNDERFS_S3_PROXY_PORT));
     }
-    if (tachyonConf.containsKey(Constants.S3_PROTOCOL)) {
-      props.setProperty("s3service.https-only", tachyonConf.get(Constants.S3_PROTOCOL)
-          .equalsIgnoreCase("http") ? "false" : "true");
+    if (tachyonConf.containsKey(Constants.UNDERFS_S3_PROXY_HTTPS_ONLY)) {
+      props.setProperty("s3service.https-only",
+          Boolean.toString(tachyonConf.getBoolean(Constants.UNDERFS_S3_PROXY_HTTPS_ONLY)));
```

Comment:
I'm doing `String` -> `Boolean` -> `String` to make sure that the user actually set the value to a boolean, and to make sure that we treat this boolean the same way we treat other booleans in Tachyon. For example, `Jets3tProperties` might error on `"True"` because it wants `"true"`, but using `tachyonConf.getBoolean()` ensures that any value which `TachyonConf.getBoolean()` considers true will be passed to `Jets3tProperties` as `"true"`. Even if `Jets3tProperties` parses `String` -> `Boolean` the same way we do, I'd rather not rely on it.